src/tsickle_compiler_host.ts

```diff
@@ -0,0 +1,259 @@
+import * as path from 'path';
+import {SourceMapConsumer, SourceMapGenerator} from 'source-map';
+import * as ts from 'typescript';
+
+import {convertDecorators} from './decorator-annotator';
+import {processES5} from './es5processor';
+import {ModulesManifest} from './modules_manifest';
+import {annotate} from './tsickle';
+
+export enum Pass {
+  None,
+  DecoratorDownlevel,
+  Tsickle
+}
+
+export interface TsickleCompilerHostOptions {
+  googmodule: boolean;
+  es5Mode: boolean;
+  tsickleTyped: boolean;
+  prelude: string;
+}
+
+export interface TsickleEnvironment {
+  /**
+   * If true, tsickle and decorator downlevel processing will be skipped for
+   * that file.
+   */
+  shouldSkipTsickleProcessing: (fileName: string) => boolean;
+  /**
+   * Takes a context (the current file) and the path of the file to import
+   *  and generates a googmodule module name
+   */
+  pathToModuleName: (context: string, importPath: string) => string;
+  /**
+   * Tsickle treats warnings as errors, if true, ignore warnings.  This might be
+   * useful for e.g. third party code.
+   */
+  shouldIgnoreWarningsForPath: (filePath: string) => boolean;
+  /**
+   * If we do googmodule processing, we polyfill module.id, since that's
+   * part of ES6 modules.  This function determines what the module.id will be
+   * for each file.
+   */
+  fileNameToModuleId: (fileName: string) => string;
+}
+
+/**
+ * TsickleCompilerHost does tsickle processing of input files, including
+ * closure type annotation processing, decorator downleveling and
+ * require -> googmodule rewriting.
+ */
+export class TsickleCompilerHost implements ts.CompilerHost {
+  private ANNOTATION_SUPPORT = `
```

Comment:
should be `private static` for a deep constant (which `CONST_CASE` implies). But prefer extracting to a local variable in the module, `const ANNOTATION_SUPPORT = ...`, no need to use a class field.